### ISSUE TITLE
File::Copy::copy(FILE, DIR) isn't documented

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#10366]
Migrated from [rt.perl.org#74974](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=74974) (status was 'resolved')

Searchable as RT74974$

**Comment by p5pRT** [comment]
### From ben@morrow.me.uk

#### Created by ben@morrow.me.uk



File&#8203;::Copy&#8203;::copy\("/some/file"\, "/an/existing/dir"\) will copy /some/file
to /an/existing/dir/file rather than failing\. While this is a useful
feature\, it isn't documented\.

<details><summary>Perl Info</summary>

```

Flags:
    category=library
    severity=low
    module=File::Copy

Site configuration information for perl 5.10.1:

Configured by mauzo at Sat Jan 16 12:52:36 GMT 2010.

Summary of my perl5 (revision 5 version 10 subversion 1) configuration:
  Commit id: 5348debf9fd57fc15c26529386769684fab96e57
  Platform:
    osname=freebsd, osvers=7.1-release-p8, archname=i386-freebsd-64int
    uname='freebsd osiris.mauzo.dyndns.org 7.1-release-p8 freebsd 7.1-release-p8 #8: sat oct 3 16:52:10 bst 2009 root@osiris.mauzo.dyndns.org:usrobjusrsrcsysosiris i386 '
    config_args='-des -Uinstallusrbinperl -Dprefix=/home/prospect/perl -Dman1dir=none -Dman3dir=none -Dinc_version_list=none -Dusemymalloc=y -Duse64bitint -Ud_dosuid'
    hint=previous, useposix=true, d_sigaction=define
    useithreads=undef, usemultiplicity=undef
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=define, use64bitall=undef, uselongdouble=undef
    usemymalloc=y, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include',
    optimize='-O',
    cppflags='-DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include -DHAS_FPSETMASK -DHAS_FLOATINGPOINT_H -fno-strict-aliasing -pipe -fstack-protector -I/usr/local/include'
    ccversion='', gccversion='4.2.1 20070719  [FreeBSD]', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=12345678
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long long', ivsize=8, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags ='-Wl,-E  -fstack-protector -L/usr/local/lib'
    libpth=/usr/lib /usr/local/lib
    libs=-lgdbm -lm -lcrypt -lutil -lc
    perllibs=-lm -lcrypt -lutil -lc
    libc=, so=so, useshrplib=false, libperl=libperl.a
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags=' '
    cccdlflags='-DPIC -fPIC', lddlflags='-shared  -L/usr/local/lib -fstack-protector'

Locally applied patches:
    


@INC for perl 5.10.1:
    /home/mauzo/src/perl/Prospect/perl/lib/5.10.1/i386-freebsd-64int
    /home/mauzo/src/perl/Prospect/perl/lib/5.10.1
    /home/mauzo/src/perl/Prospect/perl/lib/site_perl/5.10.1/i386-freebsd-64int
    /home/mauzo/src/perl/Prospect/perl/lib/site_perl/5.10.1
    .


Environment for perl 5.10.1:
    HOME=/home/mauzo
    LANG (unset)
    LANGUAGE (unset)
    LC_ALL=en_GB.UTF-8
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/mauzo/src/perl/Prospect/perl/bin:/usr/local/bin:/usr/bin:/bin:/home/mauzo/bin
    PERLDOC_PAGER=less
    PERL_BADLANG (unset)
    SHELL=/usr/local/bin/mksh

```
</details>

**Comment by p5pRT** [comment]
### From @abigail

On Fri\, May 07\, 2010 at 04&#8203;:38&#8203;:25PM \-0700\, Ben Morrow wrote&#8203;:
> 
> File&#8203;::Copy&#8203;::copy\("/some/file"\, "/an/existing/dir"\) will copy /some/file
> to /an/existing/dir/file rather than failing\. While this is a useful
> feature\, it isn't documented\.
> 


Fixed with commit 1f3ebc3b353ab41acc3013ef04412dda481b4282\.


Abigail

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @abigail

On Tue\, Jun 08\, 2010 at 04&#8203;:37&#8203;:37PM \+0200\, Abigail wrote&#8203;:
> On Fri\, May 07\, 2010 at 04&#8203;:38&#8203;:25PM \-0700\, Ben Morrow wrote&#8203;:
> > 
> > File&#8203;::Copy&#8203;::copy\("/some/file"\, "/an/existing/dir"\) will copy /some/file
> > to /an/existing/dir/file rather than failing\. While this is a useful
> > feature\, it isn't documented\.
> > 
> 
> 
> Fixed with commit 1f3ebc3b353ab41acc3013ef04412dda481b4282\.


And since it's just a doc\-patch documenting long standing behaviour\,
it should probably go into maint as well\.


Abigail

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'resolved'
